provisioning/miniprov/hare_mp/main.py

```diff
@@ -232,6 +240,28 @@ def test(args):
         shutdown_cluster()
         exit(-1)
 
+def test_IVT(args):
+    try:
+        rc = 0
+        path_to_cdf = args.file[0]
+
+        logging.info('Running test plan: ' + str(args.plan[0].value))
+        # TODO We need to handle plan type and execute test cases accordingly
+        if not is_cluster_running():
+            logging.error('Cluster is not running. Cluster must be running '
```

Comment:
This is not so obvious to me. Is it required by any specification?

Or, speaking wider, is there any requirement in which state the system will be when `test` is invoked? And in which state the system must be left when `test` stage finishes?

Reply:
@knekrasov It is expected that cluster will be already running when 'test' gets called.
We should not start/stop cluster during test stage

Reply:
So now `test_IVT` command is replacing `test` command. I believe `test()` needs to be invoked for plan `Sanity`, in that case please add that call as well.
You can have a dictionary mapping the plan to a function. Have a test function per plan, e.g. rename test() -> test_sanity(). Similarly you can define test_regression, test_functionality, etc.

Reply:
@mssawant Functions for specific plan needs to go in test rpm. So there will be file which will be installed as part of hare test rpm which will receive command from our hare_setup and will decide which functions will be called based on plan selected while executing IVT.
Let me know your thoughts.

Reply:
Don't we atleast need dummy functions or a check atleast in this case. Because, the command (hare_setup test) is part of the cortx-hare rpm but the implmentation is part of a different rpm. Now, use can see the command and when tries to execute it will fail for the given plan because test rpm is not installed.

Reply:
This problem with several RPMs can be resolved as follows:
1. Main Hare RPM provides this CLI to invoke tests. It parses the CLI parameters and extracts the "plan" name
2. All the tests are pytest-driven.
3. The tests that correspond to a certain plan can be either stored in separate folders (folder name corresponds to plan name somehow) or the plan is mapped to a [mark within the test](https://docs.pytest.org/en/6.2.x/example/markers.html).
4. Test RPM does nothing except deploying some _extended_ set of test files into the location which our `test` CLI is already aware of.

So in a nutshell `test` CLI will always invoke pytest within a given folder. The folder may be empty if test RPM is not installed.

Reply:
@mssawant We will add check similar to below once we add code for accessing hare rpm functionality.

> if checkRpm('cortx-hare-test') != 0:
            logging.error('\'cortx-hare-test\' is not installed')
            exit(-1)

Let me know your inputs

Reply:
@mssawant @knekrasov I have also raised PR for generating test rpm.
Once test rpm is installed, following commonad can be invoked from this interface
`/opt/seagate/cortx/hare/libexec/hare_test execute --plan sanity`

Reply:
As discussed in that PR `hare_test` doesn't seem to make sense